src/filters/mux_isom.c

```diff
@@ -2522,6 +2522,8 @@ static GF_Err mp4_mux_process_sample(GF_MP4MuxCtx *ctx, TrackWriter *tkw, GF_Fil
 	sap_type = gf_filter_pck_get_sap(pck);
 	if (sap_type==GF_FILTER_SAP_1)
 		tkw->sample.IsRAP = SAP_TYPE_1;
+	else if (sap_type == GF_FILTER_SAP_4)
+		tkw->sample.IsRAP = SAP_TYPE_4;
```

Comment:
I would suggest that we don't assign to SAP4 here (done later in that function), but rather assign to SAP1 in the case of non-visual media, otherwise we might break syncSample on many implementations that expect it to be always set for non visual tracks